scripts/build/chroot_generator.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package build
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func init() {
+	archiveGenerators["chroot"] = chrootGenerator{}
+}
+
+type chrootGenerator struct {
+}
+
+// The chroot generator dumps the rootfs tree into a directory appropriate for
+// running under a chroot and namespaces.
+func (g chrootGenerator) generate(config Config, files []file) error {
+	// Since files is sorted, we can guarantee that the directories are created
+	// before their children.
+	for _, f := range files {
+		f.path = filepath.Join(config.OutputPath, f.path)
+		if err := createFile(f); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Create an actual file from the file struct.
+func createFile(f file) error {
+	// TODO: set uid and gid
+	mode := fmt.Sprintf("%03o", f.mode&os.ModePerm)
+	major := fmt.Sprint(major(f.rdev))
+	minor := fmt.Sprint(minor(f.rdev))
+
+	// Special file types
+	// I'm sure many of these commands can be done through syscalls rather than
+	// forking, however we need to perform the priviledge escalation via sudo
+	// (asking for their password).
+	switch f.mode & (os.ModeType | os.ModeCharDevice) {
```

Comment:
nope. This has to change. Too many dependencies are being introduced and we've had problems in the past where they fail in different linux distros. I think it's time to suffer the pain of making system calls work. Sorry!